---
title: "Notes"
author: "Thomas Keaney, Arvid Agren and Hanna Kokko"
execute: 
  warning: false
  
---

## Project background / notes

### 25/05/2023

**Premise outline**

This project centres around the fate of alleles that encode inbreeding preference / avoidance depending on where they occur within the genome. 

But the deeper idea is that inbreeding is a 'selfish action' where some alleles benefit from helping copies of themselves in other individuals, to the detriment of the broader organism. This detriment occurs because of inbreeding depression, which occurs because of homozygosity at other loci across the genome. Homozygosity causes the expression of recessive deleterious alleles which are otherwise rarely expressed, reducing the number of viable offspring produced by the organism.

To represent the broader genome, we can use alleles on hemizygous sex chromosomes, which have no opportunity to be homozygous and thus can gain no benefits of helping a relative gain a breeding opportunity. This broadly aligns with the 'interests' of the organism. For example, in an XY system alleles on the Y do not increase the propagation of alleles identical by descent through inbreeding, because females do not carry a Y. An autosomal allele, however, does have the potential to increase the proliferation of its copies by increasing the chance of breeding with a relative. This benefit is somewhat similar to that **achieved by meiotic drive allele** (clarify your thoughts here).

A second potential angle is to explore the effect of the sex an allele encoding a preference for inbreeding is expressed in. This is important because the conditions often applicable to expression in a male present a greater parameter space for inbreeding preference (to put it too simply, because males invest less in each mating opportunity than females). Sex-specific effects also interact with hemizygous conditions, because in addition to XY systems, ZW systems exist and suddenly hemizygous chromosomes are present in females. 

So in sum, there is a likely conflict between alleles on different chromosomes over the tendency to inbreed and there is a second non-mutually exclusive conflict between the sexes over inbreeding tendency. Wow, this is cool.

This idea might be deeper still. We point out that alleles on hemizygous sex chromosomes gain no indirect inclusive fitness benefits by mating with kin. The kicker is that inbreeding is just one example of this larger idea. There are no inclusive fitness benefits for alleles that encode _any_ altruistic behaviour towards opposite sex individuals (where one sex possesses a hemizygous sex chromosome).      

**Some specific notes from the meeting**

- [Hanna's 2006 paper](https://academic.oup.com/evolut/article/60/3/467/6756586?login=true) shows when a preference for inbreeding should or should not be favoured for an autosomal allele, but does not consider other regions of the genome.

- Start with the Parker 1/3 thing, which Hanna and co re-derive. But they implicitly assume that the alleles in question here are on an autosome. We can make the change that they are on a sex chromosome, as stipulated above. 

- The X and Z can also be explored, because of their unusual inheritance dynamics. E.g. the X spends 2/3 of its time in females, but is always expressed in males and vice versa for the W. 

- Does it matter whether the loci that cause inbreeding depression are on the same/different chromosomes as those that encode inbreeding preferences...   

- Birds versus mammals, flies versus butterflies etc - different hemizygous sex chromosomes

- Unity breaks down because of inheritance differences

- Hamilton 1967 extraordinary sex chromosomes paper is a nice angle here

- Haplodiploidy - this is an extension opportunity.

- I will also try to simulate this question.

$~$

### 19/06/23

- Hanna suggests that the time in, time out continuous framework might not be the best option, as it is not the most intuitive for others. 

- Perhaps keeping it simple may be the best way forward - females mate once, males mate multiple times. This produces differences in the opportunity cost of mating between the sexes.

- A discrete time model might work better

- Scottish group models the inbreeding problem in helpful ways

- Another way is to follow the Ekrem paper and simulate in an intelligent way (this method might be particularly fruitful)

- Essentially browse the different ways to approach the problem and follow one you like best.

- Models like our developing one do not consider purging dynamics that should occur if inbreeding becomes common.

### What is relatedness exactly?

**Hamilton 1963 & 1964**

- Wright's coefficient of relationship - a pedigree based measure

**Hamilton 1971 & 1972**

- Regression coefficient between genotype of actor and recipient

**Michod and Hamilton 1980 - Coefficients of relatedness in sociobiology**

- $r = \frac{Cov(A_P, R_F)}{Cov(A_P, A_F)}$

- where $A_P$ is the genetically encoded altruistic phenotype of the actor (with values 0, 0.5, 1 corresponding to the three possible genotypes, AA, Aa and aa)

- A_F is the frequency of the altruistic genotype in the actor

- R indicates the recepient

- Orlove 1975 and Orlove and Wood 1978 should be cited here as well

**Grafen 1985 - A geometric view of relatedness**

- Crozier's relatedness (1970) = genetic similarity: the extent to which they possess the same alleles (think genomic relatedness matrices)

- Crozier's relationship (1970) = a measure of common ancestry that can be estimated from a pedigree (think animal models)

- Price's theorem is helpful. One important point is to divide the individual by its ploidy. This really matters for the inbreeding question.

**Gardner, West and Wild 2011 - The genetical theory of kin selection**

- Tend to agree with Orlove and Wood 1978: $\frac{Cov(g', g)}{Cov(g, g)}$

- Note that these are all individual measures of relatedness - we need to edit slightly to get gene's eye view measures.

## 07/11/23

- r should not be called r. It is the multiplication of two conditional probabilities. 

- The problem with current r for our purposes is that inclusive fitness is an individual level fitness metric. 

- One plot is individual level fitness loss (strategic reference gene equivalent) from inbreeding as a bar plot for optimal level of inbreeding at each chromosome

- Interlocus (or more accurately inter-chromosome) sexual conflict plot

- Haplodiploids not from the a sense but the delta sense - follow-up project?

## 26/03/24

Something to think about is how we set genetics to affect the expression of inbreeding tolerance/preference. Things that matter include:

1. do we consider inbreeding expression as the additive result of multiple loci on different chromosomes, or do we consider each case separately. **Probably not this**

2. is inbreeding expression binary, a threshold trait or a quantitative, normally distributed trait. The latter would allow us to model evolution towards some maximum. The former cases are simpler to code (particularly the first case) but still allow us to answer a very similar question. **I have this as an additive case - being heterozygous leads to expression of inbreeding preference half of the time.**

**What about paternal leakage in the mtDNA case?**

Females don't get direct benefits from inbreeding as easily as males, because they need to mate AND secure a breeding site. Breeding earlier than otherwise expected doesn't help as much.



